### PR TITLE
Paper Cutters

### PR DESCRIPTION
**Pull request by Cheridan**
cutter
cutterblade

Adds paper cutters, an office supply staple.

They cut paper.
Also people.

### CODE REVIEW

**Comment by KorPhaeron** [comment]
Doesn't fit with my design vision sorry

**Comment by Cheridan** [comment]
FUCK

File: code/modules/paperwork/paper_cutter.dm [hunk 1]
```diff
@@ -0,0 +1,94 @@
+/obj/structure/papercutter
```
Comment (Iamgoofball) [inline code/modules/paperwork/paper_cutter.dm]: >structure

>not machinery

>cant upgrade with tier 4 parts

File: code/modules/paperwork/paper_cutter.dm [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+	name = "paper cutter"
+	desc = "Standard office equipment. Precisely cuts paper using a large blade."
+	icon = 'icons/obj/bureaucracy.dmi'
+	icon_state = "papercutter-cutter"
+	var/obj/item/weapon/paper/storedpaper = null
+	var/obj/item/weapon/hatchet/cutterblade/storedcutter = null
+	var/cuttersecured = 1
```
Comment (Iamgoofball) [inline code/modules/paperwork/paper_cutter.dm]: >hardcoded securing/unsecuring

>it's not machinery

**Comment by Cheridan** [comment]
I'm not making it machinery, it doesn't need to process or use power.

File: code/modules/paperwork/paper_cutter.dm [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+	burn_state = FLAMMABLE
+	burntime = 3
+
+/obj/item/weapon/paperslip/New()
+	..()
+	pixel_x = rand(-5, 5)
+	pixel_y = rand(-5, 5)
+
+
+/obj/item/weapon/hatchet/cutterblade
```
Comment (KazeEspada) [inline code/modules/paperwork/paper_cutter.dm]: why the fuck is it a hatchet?

Comment (Cheridan) [inline code/modules/paperwork/paper_cutter.dm]: it had similar properties to what i wanted to create and none that i didn't want? typepaths don't have to follow real-world logic, they follow code reuse logic.

**Comment by ChangelingRain** [comment]
>the cutter doesn't have tablepass, so if you remove it from the table you can't put it back on

shameful

**Comment by Iamgoofball** [comment]
@Cheridan machines don't use power or process by default you dumbo

i have things that are machines for the simple fact of allowing the upgrading and construction system

File: code/modules/paperwork/paper_cutter.dm (hunk 2 above)
Comment (optimumtact) [inline code/modules/paperwork/paper_cutter.dm]: defines please

**Comment by optimumtact** [comment]
>lacks anyway to get item

**Comment by Cheridan** [comment]
i'm not making it upgradable you dingus** it's a wooden board **

**Comment by PKPenguin321** [comment]
looks pretty good to me
you could probably cut the cuttersecured var and just have the cutter go into your hands/on the floor when you use the screwdriver instead of having an arbitrary extra step, but that's up to you

File: code/modules/paperwork/paper_cutter.dm (hunk 1 above)
Comment (KazeEspada) [inline code/modules/paperwork/paper_cutter.dm]: This should be an item like paper trays.

**Comment by bgobandit** [comment]
So uh like what is the use of this besides attacking people

**Comment by optimumtact** [comment]
you can cut paper

**Comment by bgobandit** [comment]
yes but why

for what purpose

**Comment by Cheridan** [comment]
the possibilities are endless

**Comment by TheRealSnackbar** [comment]
You should be able to make paper hats and paper airplanes.

**Comment by Cheridan** [comment]
Paper hats are already in, just put paper on your head, silly.

**Comment by YotaXP** [comment]
Shouldn't wire cutters be able to make such things?  Also how do you make these?

**Comment by Jalleo** [comment]
Not in cargo crate no :moneybag: for you.

**Comment by Core0verload** [comment]
r u drunk?

**Comment by Cheridan** [comment]
@YotaXP From the sprite, I always interpreted our wirecutters as more like wirestrippers: http://cdn.xl.thumbs.canstockphoto.com/canstock22888129.jpg which couldn't cut paper at all (though other things like a knife could, but making paper strips isn't really the point, it's just a flavorful item that you can get a weapon from)

@Jalleo Guess I could add it to the bureaucracy crate. I mostly just meant it as a thing for mappers to place in appropriate office-style areas like the head of staff meeting room and the academy teaching lounge.

@Core0verload Not yet, do you want me to be?

**Comment by Razharas** [comment]
How about we add scissors instead?

**Comment by Cheridan** [comment]
holy shit fuck off

**Comment by Jalleo** [comment]
.... Why do that guys this would of been a fun gimmick for people to add stuff onto.

**Comment by Iamgoofball** [comment]
I'll remake this PR

File: code/modules/paperwork/paper_cutter.dm [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+/obj/structure/papercutter/New()
+	storedcutter = new /obj/item/weapon/hatchet/cutterblade(src)
+
+
+/obj/structure/papercutter/update_icon()
+	overlays.Cut()
+	if(!storedcutter)
+		icon_state = "papercutter"
+	else
+		icon_state = "papercutter-cutter"
```
Comment (duncathan) [inline code/modules/paperwork/paper_cutter.dm]: Could replace these with a ternary

File: code/modules/paperwork/paper_cutter.dm [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+	if(storedpaper)
+		overlays += "paper"
+	return
```
Comment (duncathan) [inline code/modules/paperwork/paper_cutter.dm]: Extraneous return

File: code/modules/paperwork/paper_cutter.dm [hunk 6]
```diff
@@ -0,0 +1,95 @@
...
+
+
+/obj/structure/papercutter/attack_hand(mob/user)
+	..()
+	src.add_fingerprint(user)
+	if(!storedcutter)
+		user << "<span class='notice'>The cutting blade is gone! You can't use \the [src] now.</span>"
+		return
+
+	if(storedcutter && !cuttersecured)
```
Comment (duncathan) [inline code/modules/paperwork/paper_cutter.dm]: You can remove the check for `storedcutter`; it returns if the above check fails

File: code/modules/paperwork/paper_cutter.dm [hunk 7]
```diff
@@ -0,0 +1,95 @@
...
+		user << "<span class='notice'>You remove [src]'s [storedcutter].</span>"
+		user.put_in_hands(storedcutter)
+		storedcutter = null
+		update_icon()
+
+	if(storedcutter && storedpaper)
```
Comment (duncathan) [inline code/modules/paperwork/paper_cutter.dm]: See above

File: code/modules/paperwork/paper_cutter.dm (hunk 3 above)
Comment (duncathan) [inline code/modules/paperwork/paper_cutter.dm]: This is fine

**Comment by duncathan** [comment]
Oh you closed it
RIP my review

**Comment by optimumtact** [comment]
Reopen cheridan you baka

**Comment by Cheridan** [comment]
IT'S AN ITEM NOW
YOU CAN BEHEAD YOURSELF WITH IT OR SMASH IT INTO YOUR STUPID FACE

ARE YOU ALL HAPPY NOW? 

:rage4:

**Comment by bgobandit** [comment]
can it cut salt

**Comment by Cheridan** [comment]
if its a big enough salt crystal then sure

**Comment by Iamgoofball** [comment]
>item

>not machinery

File: code/modules/paperwork/paper_cutter.dm [hunk 8]
```diff
@@ -0,0 +1,125 @@
...
+	var/mob/M = usr
+	if(M.incapacitated() || !Adjacent(M))
+		return
+
+	if(over_object == M)
+		M.put_in_hands(src)
+
+	else if(istype(over_object, /obj/screen))
+		switch(over_object.name)
+			if("r_hand")
```
Comment (Iamgoofball) [inline code/modules/paperwork/paper_cutter.dm]: pretty sure we have defines for these

File: code/modules/paperwork/paper_cutter.dm [hunk 9]
```diff
@@ -0,0 +1,125 @@
...
+		return (BRUTELOSS)
+	else
+		user.visible_message("<span class='suicide'>[user] repeatedly bashes [src.name] against \his head! It looks like \he's trying to commit suicide.</span>")
+		playsound(loc, 'sound/items/gavel.ogg', 50, 1, -1)
+		return (BRUTELOSS)
+
+
+/obj/item/weapon/papercutter/update_icon()
+	overlays.Cut()
+	icon_state = (storedcutter ? "papercutter-cutter" : "papercutter")
```
Comment (Iamgoofball) [inline code/modules/paperwork/paper_cutter.dm]: initial(icon_state)

File: code/modules/paperwork/paper_cutter.dm [hunk 10]
```diff
@@ -0,0 +1,125 @@
...
+	if(storedpaper)
+		overlays += "paper"
```
Comment (Iamgoofball) [inline code/modules/paperwork/paper_cutter.dm]: make this a var for subtypes

File: code/modules/paperwork/paper_cutter.dm [hunk 11]
```diff
@@ -0,0 +1,125 @@
...
+
+
+/obj/item/weapon/papercutter/attackby(obj/item/P, mob/user, params)
+	if(istype(P, /obj/item/weapon/paper) && !storedpaper)
```
Comment (Iamgoofball) [inline code/modules/paperwork/paper_cutter.dm]: hardcoded for paper, turn this into a var for future proofing/if we want metal cutters/etc.

File: code/modules/paperwork/paper_cutter.dm [hunk 12]
```diff
@@ -0,0 +1,125 @@
...
+	throwforce = 5
+	w_class = 3
+	var/obj/item/weapon/paper/storedpaper = null
+	var/obj/item/weapon/hatchet/cutterblade/storedcutter = null
+	var/cuttersecured = TRUE
+	pass_flags = PASSTABLE
+
+
+/obj/item/weapon/papercutter/New()
+	storedcutter = new /obj/item/weapon/hatchet/cutterblade(src)
```
Comment (Iamgoofball) [inline code/modules/paperwork/paper_cutter.dm]: ..()

File: code/modules/paperwork/paper_cutter.dm [hunk 13]
```diff
@@ -0,0 +1,125 @@
...
+		playsound(loc, 'sound/weapons/bladeslice.ogg', 50, 1, -1)
+		return (BRUTELOSS)
+	else
+		user.visible_message("<span class='suicide'>[user] repeatedly bashes [src.name] against \his head! It looks like \he's trying to commit suicide.</span>")
+		playsound(loc, 'sound/items/gavel.ogg', 50, 1, -1)
+		return (BRUTELOSS)
+
+
+/obj/item/weapon/papercutter/update_icon()
+	overlays.Cut()
```
Comment (Iamgoofball) [inline code/modules/paperwork/paper_cutter.dm]: ..()

File: code/modules/paperwork/paper_cutter.dm [hunk 14]
```diff
@@ -0,0 +1,125 @@
...
+	if(istype(P, /obj/item/weapon/screwdriver) && storedcutter)
+		playsound(src, 'sound/items/Screwdriver.ogg', 50, 1)
+		user << "<span class='notice'>\The [storedcutter] has been [cuttersecured ? "unsecured" : "secured"].</span>"
+		cuttersecured = !cuttersecured
+		return
+	..()
+
+
+/obj/item/weapon/papercutter/attack_hand(mob/user)
+	src.add_fingerprint(user)
```
Comment (Iamgoofball) [inline code/modules/paperwork/paper_cutter.dm]: ..()

File: code/modules/paperwork/paper_cutter.dm [hunk 15]
```diff
@@ -0,0 +1,125 @@
...
+		user << "<span class='notice'>You neatly cut \the [storedpaper].</span>"
+		storedpaper = null
+		qdel(storedpaper)
+		new /obj/item/weapon/paperslip(get_turf(src))
+		new /obj/item/weapon/paperslip(get_turf(src))
+		update_icon()
+
+
+/obj/item/weapon/papercutter/MouseDrop(atom/over_object)
+	var/mob/M = usr
```
Comment (Iamgoofball) [inline code/modules/paperwork/paper_cutter.dm]: ..()

File: code/modules/paperwork/paper_cutter.dm (hunk 12 above)
Comment (Iamgoofball) [inline code/modules/paperwork/paper_cutter.dm]: New() is important to have ..() on too

File: code/modules/paperwork/paper_cutter.dm (hunk 14 above)
Comment (Cheridan) [inline code/modules/paperwork/paper_cutter.dm]: you can't call parent here because otherwise you pick it up BECAUSE IT'S AN ITEM

**Comment by PKPenguin321** [comment]
;-;

File: code/modules/paperwork/paper_cutter.dm (hunk 11 above)
Comment (Cheridan) [inline code/modules/paperwork/paper_cutter.dm]: if someone wants to expand the item to do things i never intended they can do it themselves obv

File: code/modules/paperwork/paper_cutter.dm (hunk 15 above)
Comment (Cheridan) [inline code/modules/paperwork/paper_cutter.dm]: its fine

**Comment by PKPenguin321** [comment]
ayyy it's back

File: code/modules/paperwork/paper_cutter.dm (hunk 8 above)
Comment (duncathan) [inline code/modules/paperwork/paper_cutter.dm]: w/e

**Comment by Core0verload** [comment]
what the fuck

**Pull request merged**